extensions/wikia/FilePage/FilePageHooks.class.php

```diff
@@ -181,10 +181,14 @@ static public function onGlobalUsageLinksUpdateComplete( &$images ) {
 	 * @param User $user -- current user
 	 * @param string $reason -- undeleting reason
 	 *
-	 * @return true -- because it's hook
+	 * @return true -- because it's a hook
 	 */
 	public static function onUndeleteComplete( Title $title ) {
-		self::purgeTitle( $title );
+		global $wgRedirectFilePagesForAnons;
```

Comment:
you seem to be checking this in all hook listeners. I'm wondering - if we want to limit this to selected wikis, can we include `extensions/wikia/FilePage/FilePage.setup.php` only if `$wgRedirectFilePagesForAnons` is true?

Reply:
I tried it but unfortunately at the point of including setup only defaults are loaded and wgRedirectFilePagesForAnons is false :/